pharmd-app/src/screens/student/StudentDrawer.js

```diff
@@ -83,97 +84,113 @@ transition: ${props =>
-    isOpen
-      ? setStudentQuickViewExpanded(!studentQuickViewExpanded)
-      : setStudentQuickViewExpanded(true);
-  };
-
-  // Avoid route errors
-  const quickview = () => {
-    return isOpenMatch ? (
-      <>
-        <StudentQuickView id={selected} onCancel={handleClose} {...props} />
-        <ButtonSpan>
-          <RouterLink to={`/students/${props.id}/details`}>
-            <DetailsButton>More Student Info</DetailsButton>
-          </RouterLink>
-        </ButtonSpan>
-      </>
-    ) : (
-      <div>No Student selected</div>
-    );
-  };
-
-  return (
-    <Drawer variant="permanent" open={isDrawerOpen} anchor="right" onClose={handleClose}>
-      <NavItemSecondary
-        title={isDrawerOpen ? "Close" : "Open"}
-        iconSrc={VerticalSplitIcon}
-        onClick={isDrawerOpen ? handleClose : handleOpen}
-        isOpen={isDrawerOpen}
-        isActive={isDrawerOpen}
-      />
-      <ExpansionPanel
-        SummaryChild={
-          <NavItemSecondary
-            title="Table Filters"
-            iconSrc={FilterIcon}
-            onClick={handleOpen}
-            isOpen
-            isActive={false}
-            sidebarIsOpen={isDrawerOpen}
-          />
-        }
-        DetailChild={<StudentDrawerFilter {...useListController(props)} />}
-        defaultExpanded={isDrawerOpen}
-        expanded={isDrawerOpen && filtersQuickViewExpanded}
-        onChange={changeFiltersExpansionPanel}
-      />
-      <ExpansionPanel
-        SummaryChild={
-          <>
+const StudentDrawer = ({ isOpenMatch, selected, handleClose, handleOpen, ...props }) => {
+
+    let isDrawerOpen = props.studentSidebar;
+
+    const [filtersQuickViewExpanded, setFiltersQuickViewExpanded] = useState(false);
+
+    // onChange functions for when the expansion panel is clicked on:
+    // The toggles only toggle when the sidebar is open, otheriwse they close
+    const toggleFiltersExpansionPanel = () => {
+      if(isDrawerOpen){
+        setFiltersQuickViewExpanded(!filtersQuickViewExpanded)
+      }
+      else{
+        setFiltersQuickViewExpanded(true);
+        openSidebar();
+      }
+
+    };
+
+
+    const toggleStudentExpansionPanel = () => {
+      if(isDrawerOpen){
+        props.setStudentQuickViewExpanded(!props.studentQuickViewExpanded)
+      }
+      else{
+        props.setStudentQuickViewExpanded(true);
+        openSidebar();
+      }
+    };
+
+    const closeSidebar = () => {
+      props.setStudentSidebar(false);
+    }
+
+    const openSidebar = () => {
+      props.setStudentSidebar(true);
+    }
+
+
+    const toggleStudentSidebar = () => {
+      props.setStudentSidebar(!props.studentSidebar);
+    }
+
+
+
+// Avoid route errors
+    const quickview = () => {
+      return isOpenMatch ? (
+        <>
+          <StudentQuickView id={selected}  {...props} />
+          <ButtonSpan>
+            <RouterLink to={`/students/${props.id}/details`}>
+              <DeatilsButton>More Student Info</DeatilsButton>
```

Comment:
There's a typo here. It should be `DetailsButton`. Otherwise, everything still works!